doc/SONiC_local_users_password_reset_hld.md

```diff
@@ -0,0 +1,300 @@
+
+
+def main():
+    LocalUsersConfigurationReset().start()
+```
+
+The feature is added to the init flow as long as we set the new ```ENABLE_LOCAL_USERS_PASSWORDS_RESET``` global variable and the exsiting ```CHANGE_DEFAULT_PASSWORD``` global variable in ```rules/config``` to ```y```.
+
+```
+# ENABLE_LOCAL_USERS_PASSWORDS_RESET - enable local users' passwords reset during init on switch
+ENABLE_LOCAL_USERS_PASSWORDS_RESET ?= y
+```
+
+###
+
+# Performance
+
+The ```reset-local-users-passwords.service``` is required to run as fast as possible, and it is expected to finish running in milliseconds **(150-300 milliseconds)** on long reset detected, and not delay the init flow more than that.
+
+# Flows
+
+###### Local users' password reset during init flow
+![](./reset-local-users-passwords-init-flow.png  "Local users' password reset during init")
+
+The defualt behavior is on long reset button press, the system will delete non default users and restore original passwords for default users and expire them, all done using linux commands
+
+# Configuration and management
+
+## CLI
+
+<!-- omit in toc -->
+## Command structure
+
+**User interface**:
+```
+config
+\-- local-users-passwords-reset
+    |-- state <enabled|disabled>
+
+show
+\-- local-users-passwords-reset
+```
+
+### Config command group
+**The following command set local-users-passwords-reset  feature state:**
+```bash
+config local-users-passwords-reset state <enabled|disabled>
+```
+
+### Show command
+**The following command display long-reset-button-press  configuration:**
+```bash
+root@sonic:/home/admin# show local-users-passwords-reset
+state
+-------
+enabled
+```
+
+## YANG model
+New YANG model `local-users-passwords-reset.yang` will be added to provide support for configuring llocal-users-passwords-reset state.
+
+**Skeleton code:**
+```
+ module sonic-local-users-passwords-reset {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/local-users-passwords-reset";
+
+    description "LONG_RESET_BUTTON YANG Module for SONiC-based OS";
+
+    revision 2024-01-04 {
+        description "First Revision";
+    }
+
+    container sonic-local-users-passwords-reset {
+
+        container LOCAL_USERS_PASSWORDS_RESET {
+
+            description "LOCAL_USERS_PASSWORDS_RESET part of config_db.json";
+
+            container STATE {
+                leaf state {
+                    type string {
+                        pattern "enabled|disabled";
+                    }
+                    description "Local users' password reset during init feature state";
+                    default disabled;
+                }
+
+            } /* end of container STATE */
+        }
+        /* end of container LOCAL_USERS_PASSWORDS_RESET */
+    }
+    /* end of top level container */
+}
+/* end of module sonic-local-users-passwords-reset */
+```
+
+
+# Warmboot and Fastboot Design Impact
+
+The feature will not have any impact on fast reboot or warmboot.
```

Comment:
This daemon itself shouldn't run when the boot type is warmboot or fastboot as the reset button press results only in cold boot. Warmboot and fastboot have strict timing requirements and since this flow will not happen during warm or fastreboot, lets make sure not to run the daemon. 
Please update the same in HLD as well

Reply:
I will update it also in the requirements